main.cpp

```diff
@@ -31,9 +31,129 @@
 
  Wait for my code review.
  */
+ //FloatType
+struct FloatType
+{
+    float add( float, float );
+    float subtract( float, float );
+    float multiply( float, float );
+    float divide( float, float );
+};
+
+float FloatType::add(float x, float y)
+{
+    return x + y;
+}
+
+float FloatType::subtract(float x, float y)
+{
+    return x - y;
+}
+float FloatType::multiply(float x, float y)
+{
+    return x * y;
+}
+float FloatType::divide(float x, float y)
+{
+    return x / y;
+}
+
+//DoubleType
+struct DoubleType
+{
+    double add( double, double );
+    double subtract( double, double );
+    double multiply( double, double );
+    double divide( double, double );
+};
+
+double DoubleType::add( double x, double y )
+{
+    return x + y;
+}
+
+double DoubleType::subtract( double x, double y )
+{
+    return x - y;
+}
+
+double DoubleType::divide( double x, double y )
+{
+    return x / y;
+}
+
+double DoubleType::multiply( double x, double y )
+{
+    return x * y;
+}
+
+//IntType
+struct IntType
+{
+    int add( int, int );
+    int subtract( int, int );
+    int multiply( int, int );
+    int divide( int, int );
+};
+
+int IntType::add( int x, int y )
+{
+    return x + y;
+}
+
+int IntType::subtract( int x, int y )
+{
+    return x - y;
+}
+
+int IntType::divide( int x, int y )
+{
+    return x / y; 
+}
```

Comment:
When you divide an integer by 0, the program will crash.  Floating point division-by-zero does not crash the program.  You need to add code to handle this case here. 
in pseudo-code, that would look like this:
```
if y equals 0
    "warning, dividing by 0!"
    return 0

return x / y
```

- [ ] revise function to handle when `y` is 0